python/node_classification.py

```diff
@@ -0,0 +1,744 @@
+                current_values[OtherParams.DEVICE_TYPE],
+            )
+            logged_data.append(
+                {
+                    "epoch": epoch,
+                    "loss": loss,
+                    "val_loss": val_loss,
+                    "train": dict_train,
+                    "val": dict_val,
+                }
+            )
+
+            print(
+                f'Epoch: {epoch:03d}, Loss: {loss:.4f}, Val Loss: {val_loss:.4f}, Accuracy: {logged_data[-1]["train"]["accuracy"]:.4f}, Accuracy: {logged_data[-1]["val"]["accuracy"]:.4f}'
+            )
+
+        # save model every checkpoint_freq epochs
+        if epoch % current_values[TrainParams.CHECKPOINT_FREQ] == 0:
+            if (
+                time() - last_time
+                > current_values[TrainParams.TIME_BETWEEN_CHECKPOINTS]
+            ):
+                save_model_to_folder()
+                last_time = time()
+
+    return [
+        mgp.Record(
+            epoch=data["epoch"],
+            loss=data["loss"],
+            val_loss=data["val_loss"],
+            train_log=data["train"],
+            val_log=data["val"],
+        )
+        for data in logged_data
+    ]
+
+
+##############################
+# get training data
+##############################
+
+
+@mgp.read_proc
+def get_training_data() -> mgp.Record(
+    epoch=int, loss=float, val_loss=float, train_log=mgp.Any, val_log=mgp.Any
+):
+    """This function is used so user can see what is logged data from training.
+
+
+    Returns:
+        mgp.Record(
+            epoch (int): epoch number of record of logged data row
+            loss (float): loss in logged data row
+            val_loss (float): validation loss in logged data row
+            train_log (mgp.Any): training parameters of record of logged data row
+            val_log (mgp.Any): validation parameters of record of logged data row
+            ): record to return
+
+
+    """
+
+    return [
+        mgp.Record(
+            epoch=logged_data[k]["epoch"],
+            loss=logged_data[k]["loss"],
+            val_loss=logged_data[k]["val_loss"],
+            train_log=logged_data[k]["train"],
+            val_log=logged_data[k]["val"],
+        )
+        for k in range(len(logged_data))
+    ]
+
+
+##############################
+# model loading and saving, predict
+##############################
+
+
+@mgp.read_proc
+def save_model() -> mgp.Record(path=str, status=str):
+    """This function saves model to model saving folder. If there are already total
+    of max_models_to_keep models in model saving folder, oldest model is deleted.
+
+    Exception: raised if model is not initialized or defined
+
+    Returns:
+        mgp.Record(
+            path (str): path to saved model
+            status (str): status of saving model
+            ): return record
+    """
+
+    if model is None:
+        raise Exception(
+            "Saving is not enabled until model is not initialized or loaded."
```

Comment:
The current wording can easily be interpreted in the opposite way (Croatian is weird with "until" and negation)
```suggestion
            "There are no initialized or loaded models."
```

```suggestion
            "First load or initialize a model to be able save it."
```

Reply:
agreed